### ISSUE TITLE
Make line split regex ignore escaped quotes

### ISSUE THREAD

**Issue by Katsute** [Katsute/MyAnimeList-Java-API#22]
The current line split regex works except in cases where an escaped quote exists. This needs to be fixed so that it works with escaped quotes.

```regex
(?<=[{\[,]|(?=[}\]])) ?(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)(?!$)
```

Example where regex works: `{"obj":{},"arr":[],"escaped":"str"}`
Example where regex fails: `{"obj":{},"arr":[],"escape\"d":"str"}`

The fixed regex should split where there is a `{}[],` except within quotes.

**Do not make a pull**, instead post a new regex in the comments

**Comment by Katsute** [comment]
This regex seems to work:

```regex
(?<=[{\[,]|(?=[}\]]))()(?:\"[^\"]*\"|[^,]+)
```

**Comment by Katsute** [comment]
Line split regex is causing stack overflow exception due to its inefficiency

**Comment by Katsute** [comment]
A slower, manual approach may be required for line splitting (char by char?)

**Comment by Katsute** [comment]
Potential fix:

psuedo:
```
StringBuilder OUT;

int index = 0;

(char) for each occurance of {}[],
	(int) count non-escaped quotes after 
	
	if ^ EVEN :  // odd number implies inside quotes
		OUT += substring(index, match index) // everything after previous match
		
		index = match index
		
		if char = {[, :  // append newline after
			OUT += '\n'
			OUT += char
		else char = }] :  // append new line before
			OUT += char
			OUT += '\n'
```

**Comment by Katsute** [comment]
Fixed in 4d18d3944ea791dc9d5ab44988e78839cf9b8027
